Sources/Sentry/SentryANRTracker.m

```diff
@@ -68,7 +63,7 @@ - (void)detectANRs
         if (blockExecutedOnMainThread) {
```

Comment:
Note: while this hasn't changed in this PR, I wanted to point out the issue with this approach (the same one is present in the Java watchdog AFAIK:

Sleeping for a given duration before checking a flag means the timeout is not really a "timeout", but merely a baselin, with the actual UI non-responsive time necessary for the ANR to be reported ranging between `timeoutInterval` & `timeoutInterval * 2`, averaging at about `timeoutInterval * 1.5`.  That's because the following happens:

Assume timeoutInterval = 2 seconds, i.e. the current default. Also let's call `blockExecutedOnMainThread` just `flag` for brewity:
1. `flag= NO` is executed and a block to reset is is scheduled
2. say 0.1 second later, the `flag = YES` is executed on the main thread because UI still works
3. say another 0.1 second later, UI hangs. Current time is `start+0.2 seconds`
4. at `start+2 seconds` the watchdog thread wakes up, sees everything working just fine and looping. UI is still stuck at this point but `flag == YES`...
5. in another iteration, set `flag = NO`
6. now the UI starts being responsive after a total of 3.7 second unresponsiveness, with the current time being at `start + 3.9 second`, i.e. just before the next check. Sets `flag = YES`, since UI works now again.
8. next check happens after the two-second sleep, all nice and shiny again because `flag == YES` - no ANR detected.

I'm not saying this is a huge deal, just pointing out the timeout isn't really a timeout... That's why I've changed this in [the Unity implementation](https://github.com/getsentry/sentry-unity/pull/771/files#diff-c0f681085f84da731abf24428667d8fa6285481ebb12bffdf537fa4592ce1c06R122-R142) (after realizing this issue because my tests were spuriously failing), to executing the background iteration more often (5 times more often, currently) and counting the number of iterations (`ticks`) since the UI last responded. Then if the count reaches the predetermined count (5), the report is sent. Still not exact, but reduces the average ANR time before report from `timetout*1.5` to `timeout * 1.1` and more importantly, but more importantly, it also reduces the maximum time when an ANR **isn't detected** to about `timeout*1.2` instead of `timeout*1.9999...`

Reply:
Thanks. I will open an issue to deal with this on  another PR.

Reply:
https://github.com/getsentry/sentry-cocoa/issues/1865

Reply:
Thanks for pointing that out @vaind. Yes, I'm aware of that problem. Implementing an improved version of checking the background iteration more often would do the trick. I should have documented that we initially decided against this approach because it's a bit more complex. I wanted to ship something quickly to get rid of falsely reported OOMs.